### DISCUSSION TITLE
Convert temperature sensors in Kelvin to configured temperature unit

### DISCUSSION THREAD

**Discussion by Flameeyes** [home-assistant/architecture#665] [Entity Models] [answered]
Currently, temperature sensors in Celsius and Fahrenheit are converted to the configured temperature unit for consistency across UI (as  described in https://developers.home-assistant.io/docs/core/entity/sensor/ .)

There are sensors that report values in Kelvin, rather than C/F, but still represent a temperature — in particular that is the case for Dyson fan reported temperature, but I see that Tasmota, ISY994, and LCN integration at least theoretically support that concept.

The conversion likely cannot be applied to all sensors with a Kelvin unit though, as color temperature is typically represented in Kelvin, and would be confusing to be represented in C/F (I have not found any sensor that appears to use it this way within core, but I can see the usage of it to expose white temperature of lights with template sensors.)

**Comment by frenck** [comment]
My personal feeling, it that we should leave this up to the integration to resolve and not stuff this into the sensor entity component.

The main reason is for the same thing you already mentioned: Kelvin has other applications, e.g., color temperature. 

IMHO, cases of devices reporting Kelvin for air temperature, is rare, and should not be a case we should solve in our entity-component for that reason.

**Reply by Flameeyes** [reply]
Totally agreed it's a rare corner case, though in my experience so is using color temperature sensors.

I guess the one thing that I don't know is whether color temperature should also be using `DEVICE_CLASS_TEMPERATURE` — if so, then yeah I would abandon this change for sure.

If not, while it might be overkill to solve in the entity-component, the changes needed for it for the sake of completeness are fairly minimal, particularly if we wait for the removal of conversion for non-device-class-tagged sensors — they are the only complications in https://github.com/home-assistant/core/pull/59703 as the K→C and K→F conversion is already implemented.

**Comment by Flameeyes** [comment, marked as answer]
Does https://www.home-assistant.io/blog/2022/04/06/release-20224/#adjusting-unit-of-measurement-for-sensors mean that this is now changing? I see K next to C/F in the notes…

**Reply by bdraco** [reply]
I'm pretty sure this is actually implemented via https://github.com/home-assistant/core/pull/64366
